make all three ADCs available to the user

mazzoo opened the pull request:
with a simple dt overlay

&adc1 {
    status = "okay";
};

she can enable one ADC at a time

martinjaeger commented on dts/arm/st/f7/stm32f7.dtsi:
```diff
@@ -634,14 +634,34 @@
 		adc1: adc@40012000 {
 			compatible = "st,stm32-adc";
-			reg = <0x40012000 0x400>;
+			reg = <0x40012000 0x50>;
```
For some reason the length is always set to `0x400`. Maybe @erwango can knows what's the reason behind this?

mazzoo replied:
the per-ADC regspace is only 0x50, but there are four common registers related to all three of the ADCs. luckily the adc_stm32 driver does not access this common configuration space, which makes my hack work.



![ADC_regs](https://user-images.githubusercontent.com/151651/106269317-a61e0380-622c-11eb-9871-a0e182c03136.png)

erwango replied:
@martinjaeger The reg space per device instance is typically 0x400, so copy/paste after copy/paste it ended up here and since this value is not used in the code, no one saw this was not correct in this case.

martinjaeger replied:
Ok, thanks. So it's good to use the correct space of `0x050` here.

martinjaeger commented on dts/arm/st/f7/stm32f7.dtsi:
```diff
@@ -634,14 +634,34 @@
 			#io-channel-cells = <1>;
 		};
 
+		adc2: adc@40012000 {
```
This needs to have the same address as the `reg` node below.
```suggestion
		adc2: adc@40012100 {
```

mazzoo replied:
each of the three ADC only has 0x50 as its own address space. however there are three "common registers". luckily the adc_stm32 driver doesn't use these three common registers.

![ADC_regs](https://user-images.githubusercontent.com/151651/106279663-0d43b400-623d-11eb-8f25-bc92d5c27765.png)

martinjaeger reviewed (changes requested):
Thanks @mazzoo for this contribution.

Please see [here](https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines) for documentation how to fix the commit message such that it passes CI. It should start with e.g. `dts: arm: stm32f7: ...`.

mwadva committed:
dts: arm: stm32f7: make all three ADCs available to the user

with a simple dt overlay

&adc1 {
status = "okay";
};

user can enable one ADC at a time. but only one, as for multiple ADCs
the interrupts would collide.

Signed-off-by: Matthias Wenzel <mw@adva.store>
